### PR TITLE
Update framework.md with example application.

### PR DESCRIPTION
**Pull request by grasmash**
This pull request adds an example application to framework.md

### CODE REVIEW

File: docs/framework.md [hunk 1]
```diff
@@ -76,14 +76,79 @@ Pass the resulting `$commandClasses` to the `Runner()` constructor as shown abov
...
+use Psr\Log\LoggerAwareTrait;
+use Robo\Common\ConfigAwareTrait;
+use Robo\Config;
+use Robo\Robo;
+use Robo\Runner as RoboRunner;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MyApplication implements ContainerAwareInterface, LoggerAwareInterface {
```
Comment (greg-1-anderson) [inline docs/framework.md]: I think for the example, it is not necessary to have the Application depend on ContainerAwareInterface or LoggerAwareInterface. Encourage folks to do most of their work outside the application.

Comment (greg-1-anderson) [inline docs/framework.md]: Folks who want to log or manage objects inside their application can, of course, add these if they are needed.

File: docs/framework.md [hunk 2]
```diff
@@ -76,14 +76,79 @@ Pass the resulting `$commandClasses` to the `Runner()` constructor as shown abov
...
+```php
+<?php
 use League\Container\Container;
 
 $input = new \Symfony\Component\Console\Input\ArgvInput($argv);
 $output = new \Symfony\Component\Console\Output\ConsoleOutput();
 $conf = new \Robo\Config(); \\ or use your own subclass
-$app = new \My\Application();
+$app = new \MyApplication($config, $input, $output);
 $container = \Robo\Robo::createDefaultContainer($input, $output, $app, $conf);
```
Comment (greg-1-anderson) [inline docs/framework.md]: Did you mean to delete this line? There's a createDefaultContainer below as well.

File: docs/framework.md [hunk 3]
```diff
@@ -76,14 +76,79 @@ Pass the resulting `$commandClasses` to the `Runner()` constructor as shown abov
...
+    $container = Robo::createDefaultContainer($input, $output, $application,
+      $config);
+    $this->setContainer($container);
+    $container->add(MyCustomService::class);
+
+    // Add commands.
+    $this->commands = [My\Custom\Command::class];
+
+    // Instantiate Robo Runner.
+    $this->runner = new RoboRunner();
```
Comment (greg-1-anderson) [inline docs/framework.md]: Just pass `[My\Custom\Command::class]` as the constructor to the runner. Terminus does some more complicated stuff, but it is sufficient for many folks to just pass the list of command classes to the runner, and dispense with `$this->commands` et. al.

File: docs/framework.md [hunk 4]
```diff
@@ -76,14 +76,79 @@ Pass the resulting `$commandClasses` to the `Runner()` constructor as shown abov
...
+    $this->runner->setContainer($container);
+
+    $this->setLogger($container->get('logger'));
+  }
+
+  public function run(InputInterface $input, OutputInterface $output) {
+    $status_code = $this->runner->run($input, $output, NULL, $this->commands);
```
Comment (greg-1-anderson) [inline docs/framework.md]: This can just be `$this->runner->run($input, $output);`

**Comment by greg-1-anderson** [comment]
Thanks for the documentation contribution -- very helpful!

**Comment by grasmash** [comment]
@greg-1-anderson I've updated according to your feedback.

**Comment by greg-1-anderson** [comment]
Thanks very much for this!

**Pull request merged**